File: sdk/javascript/examples/quickstart.mjs
// Démarrage rapide — node examples/quickstart.mjs (après `npm run build`)
import { LoyaltyClient, ApiError, SignatureVerificationError } from "../dist/index.js";

// 1. Initialisation — récupérez ces clés dans Portail > Applications
const loyalty = new LoyaltyClient(
    process.env.LOYALTY_PUBLIC_KEY ?? "pk_test_xxx",
    process.env.LOYALTY_PRIVATE_KEY ?? "sk_test_xxx",
    process.env.LOYALTY_BASE_URL ?? "http://localhost:8081",
    { webhookSecret: process.env.LOYALTY_WEBHOOK_SECRET }
);

// 2. Envoyer un événement d'achat (avec clé d'idempotence)
try {
    const result = await loyalty.trackEvent("purchase.completed", "00000000-0000-0000-0000-000000000123", {
        payload: { amount: 4990 },
        idempotencyKey: "commande-2026-0042",
    });
    console.log("Événement traité :", result.eventId);
    for (const effect of result.effectsApplied) {
        console.log(` → effet : ${effect.effectType} (règle : ${effect.ruleName})`);
    }
} catch (e) {
    if (e instanceof ApiError) console.error(`Erreur API (${e.statusCode}) : ${e.message}`);
    else throw e;
}

// 3. Consulter le solde de points du membre
const points = await loyalty.getMemberPoints("00000000-0000-0000-0000-000000000123");
console.log("Points disponibles :", points.availablePoints);

// 4. Dans votre route de callback Express :
//    app.post("/loyalty/callback", express.raw({ type: "application/json" }), (req, res) => {
//      try {
//        const event = loyalty.checkCallbackIntegrity(req.headers, req.body);
//        res.status(200).send("OK");
//      } catch (e) {
//        if (e instanceof SignatureVerificationError) res.status(400).send("KO");
//        else throw e;
//      }
//    });

File: sdk/python/examples/quickstart.py
"""Démarrage rapide — python examples/quickstart.py"""

import os

from loyalty_sdk import ApiError, LoyaltyClient

# 1. Initialisation — récupérez ces clés dans Portail > Applications
loyalty = LoyaltyClient(
    public_key=os.environ.get("LOYALTY_PUBLIC_KEY", "pk_test_xxx"),
    private_key=os.environ.get("LOYALTY_PRIVATE_KEY", "sk_test_xxx"),
    base_url=os.environ.get("LOYALTY_BASE_URL", "http://localhost:8081"),
    webhook_secret=os.environ.get("LOYALTY_WEBHOOK_SECRET"),
)

MEMBER_ID = "00000000-0000-0000-0000-000000000123"

# 2. Envoyer un événement d'achat (avec clé d'idempotence)
try:
    result = loyalty.track_event(
        "purchase.completed",
        MEMBER_ID,
        payload={"amount": 4990},
        idempotency_key="commande-2026-0042",
    )
    print("Événement traité :", result["eventId"])
    for effect in result["effectsApplied"]:
        print(f" → effet : {effect['effectType']} (règle : {effect['ruleName']})")
except ApiError as error:
    print(f"Erreur API ({error.status_code}) : {error}")

# 3. Consulter le solde de points du membre
points = loyalty.get_member_points(MEMBER_ID)
print("Points disponibles :", points.get("availablePoints"))

# 4. Dans votre vue de callback (ex. Flask) :
#    @app.post("/loyalty/callback")
#    def loyalty_callback():
#        try:
#            event = loyalty.check_callback_integrity(request.headers, request.get_data())
#            return "OK", 200
#        except SignatureVerificationError:
#            return "KO", 400
